subsys/nrf_security/src/drivers/cracen/cracenpsa/src/key_management.c

```diff
@@ -649,6 +649,22 @@ static psa_status_t handle_curve_family(psa_ecc_family_t psa_curve, size_t key_b
 	return PSA_SUCCESS;
 }
 
+static bool requires_sitask(const psa_key_attributes_t *attributes, psa_ecc_family_t curve)
+{
+	if (!(IS_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_MONTGOMERY_255) ||
+	      IS_ENABLED(PSA_NEED_CRACEN_KEY_TYPE_ECC_MONTGOMERY_448) ||
+	      IS_ENABLED(PSA_NEED_CRACEN_ECDSA_SECP_R1_256))) {
+		return false;
+	}
+	if ((curve != PSA_ECC_FAMILY_TWISTED_EDWARDS && curve != PSA_ECC_FAMILY_SECP_R1 &&
+		curve != PSA_ECC_FAMILY_SECP_K1 && curve != PSA_ECC_FAMILY_BRAINPOOL_P_R1) ||
+	    (PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
+	     PSA_KEY_LOCATION_CRACEN)) {
+		return true;
+	}
+	return false;
```

Comment:
Here you could also just
```suggestion
	return (curve != PSA_ECC_FAMILY_TWISTED_EDWARDS && curve != PSA_ECC_FAMILY_SECP_R1 &&
		curve != PSA_ECC_FAMILY_SECP_K1 && curve != PSA_ECC_FAMILY_BRAINPOOL_P_R1) ||
	    PSA_KEY_LIFETIME_GET_LOCATION(psa_get_key_lifetime(attributes)) ==
	     PSA_KEY_LOCATION_CRACEN;
```

Reply:
True, but I am, thankfully, able to remove most of this in another pr soon which will make this a lot cleaner